test/integration/scheduler/queue_test.go

```diff
@@ -1290,6 +1298,640 @@ func TestCoreResourceEnqueue(t *testing.T) {
 			wantRequeuedPods:          sets.New("pod2"),
 			enableSchedulingQueueHint: []bool{true},
 		},
+		{
+			name:         "Pod rejected with node by the VolumeBinding plugin is requeued when the Node is created",
+			initialNodes: []*v1.Node{st.MakeNode().Name("fake-node").Label("node", "fake-node").Obj()},
+			initialPVs: []*v1.PersistentVolume{
+				st.MakePersistentVolume().Name("pv1").
+					AccessModes([]v1.PersistentVolumeAccessMode{v1.ReadOnlyMany}).
+					Capacity(v1.ResourceList{v1.ResourceStorage: resource.MustParse("1Mi")}).
+					HostPathVolumeSource(&v1.HostPathVolumeSource{Path: "/tmp", Type: ptr.To(v1.HostPathDirectoryOrCreate)}).
+					Obj(),
+			},
+			initialPVCs: []*v1.PersistentVolumeClaim{
+				st.MakePersistentVolumeClaim().
+					Name("pvc1").
+					AccessModes([]v1.PersistentVolumeAccessMode{v1.ReadWriteMany}).
+					Resources(v1.VolumeResourceRequirements{Requests: v1.ResourceList{v1.ResourceStorage: resource.MustParse("2Mi")}}).
+					StorageClassName(ptr.To("wait-sc")).
+					Obj(),
+			},
+			pods: []*v1.Pod{
+				st.MakePod().Name("pod1").Container("image").PVC("pvc1").Obj(),
+			},
+			triggerFn: func(testCtx *testutils.TestContext) (map[framework.ClusterEvent]uint64, error) {
+				node := st.MakeNode().Name("fake-node2").Label("node", "fake-node").Obj()
```

Comment:
Is this new node giving any possibility for the pod to be scheduled on it? Ignore the fact we have always Queue hint for NodeAdd event. Let's make the test case more real, so create init node on which the pod cannot be scheduled and then create a new one where it can be scheduled.

Reply:
Similarly you can make NodeUpdate.

Reply:
Thank you for the review. By adding NodeAffinity to the PV, init node on which the pod cannot be scheduled and then create a new one where it can be scheduled. The same applies to the NodeUpdate case as well.